### PR TITLE
Change ApplicationProtocol to ReadOnlyMemory<byte> 

### PR DESCRIPTION
**Pull request by Tratcher**
#2182 
In the process I realized the prior change for merging the https and core packages had dropped the new netcoreapp2.1 TFM so all of the HTTP/2 code was disabled. Filed https://github.com/aspnet/KestrelHttpServer/issues/2238 for followup.

### CODE REVIEW

File: samples/Http2SampleApp/Http2SampleApp.csproj [hunk 1]
```diff
@@ -15,7 +15,9 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Content Include="../../test/shared/TestCertificates/testCert.pfx" CopyToOutputDirectory="PreserveNewest" />
+    <Content Include="testCert.pfx">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (halter73) [inline samples/Http2SampleApp/Http2SampleApp.csproj]: Out of curiosity, does CopyToOutputDirectory need to be a tag, or does it still work as a attribute?

File: src/Kestrel.Core/Kestrel.Core.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <AssemblyName>Microsoft.AspNetCore.Server.Kestrel.Core</AssemblyName>
     <RootNamespace>Microsoft.AspNetCore.Server.Kestrel.Core</RootNamespace>
     <Description>Core components of ASP.NET Core Kestrel cross-platform web server.</Description>
-    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (halter73) [inline src/Kestrel.Core/Kestrel.Core.csproj]: How was this working before without the 2.1 target?

File: src/Kestrel.Core/Internal/HttpConnection.cs [hunk 3]
```diff
@@ -303,7 +305,8 @@ private void DisposeAdaptedConnections()
         private HttpProtocols SelectProtocol()
         {
             var hasTls = _context.ConnectionFeatures.Get<ITlsConnectionFeature>() != null;
-            var applicationProtocol = _context.ConnectionFeatures.Get<ITlsApplicationProtocolFeature>()?.ApplicationProtocol;
+            var applicationProtocol = _context.ConnectionFeatures.Get<ITlsApplicationProtocolFeature>()?.ApplicationProtocol
```
Comment (halter73) [inline src/Kestrel.Core/Internal/HttpConnection.cs]: Can't we now just get rid of ITlsApplicationProtocolFeature and pull the application protocal from the ITlsConnectionFeature we access a line above?

File: src/Kestrel.Core/Kestrel.Core.csproj (hunk 2 above)
Comment (Tratcher) [reply src/Kestrel.Core/Kestrel.Core.csproj]: It wasn't. See the PR summary.

File: src/Kestrel.Core/Internal/HttpConnection.cs (hunk 3 above)
Comment (Tratcher) [reply src/Kestrel.Core/Internal/HttpConnection.cs]: ITlsConnectionFeature is from HttpAbstractions, it's a breaking change to merge them.

File: samples/Http2SampleApp/Http2SampleApp.csproj (hunk 1 above)
Comment (Tratcher) [reply samples/Http2SampleApp/Http2SampleApp.csproj]: It might work as an attribute but this is what VS consistently generates.

Comment (davidfowl) [reply samples/Http2SampleApp/Http2SampleApp.csproj]: But it looks cleaner as an attribute 😄

File: src/Kestrel.Core/Internal/HttpConnection.cs (hunk 3 above)
Comment (Tratcher) [reply src/Kestrel.Core/Internal/HttpConnection.cs]: Oh, I see what you mean. The implementation is already merged but we're accessing it via interfaces and can't tell.

Comment (halter73) [reply src/Kestrel.Core/Internal/HttpConnection.cs]: Just delete ITlsApplicationProtocolFeature.cs

Comment (halter73) [reply src/Kestrel.Core/Internal/HttpConnection.cs]: I see. ITlsConnectionFeature itself doesn't include ApplicationProtocol though the implementation does. Just get rid of the TODO then.

**Pull request merged**